src/internal/observable/partition.ts

```diff
@@ -4,15 +4,15 @@ import { ObservableInput } from '../types';
 import { Observable } from '../Observable';
 import { innerFrom } from './from';
 
-export function partition<T, U extends T, A>(
+export function partition<T, U extends T, A, S extends Exclude<T, U>>(
   source: ObservableInput<T>,
   predicate: (this: A, value: T, index: number) => value is U,
   thisArg: A
-): [Observable<U>, Observable<T>];
-export function partition<T, U extends T>(
+): [Observable<U>, Observable<S>];
+export function partition<T, U extends T, S extends Exclude<T, U>>(
   source: ObservableInput<T>,
   predicate: (value: T, index: number) => value is U
-): [Observable<U>, Observable<T>];
+): [Observable<U>, Observable<S>];
```

Comment:
AFAICT, `S` doesn't have to be a type parameter. Instead, `Observable<Exclude<T, U>>` could be used in the return type.